src/plugins/BehaviorSpecGenerator/BehaviorSpecGenerator.js

```diff
@@ -254,28 +267,237 @@ define([
+                        totalStateNames[childName] = this.core.getPath(child);
+                    }
+
+                    if ((this.isMetaTypeOf(child, this.META.EnforceableTransition)) || (this.isMetaTypeOf(child, this.META.SpontaneousTransition))) {
+
+                        if (this.core.getPointerPath(child, 'dst') === null) {
+                            violations.push({
+                                    node: node,
+                                message: 'Dst of connector [' + childPath + '] is null'
+                            });
+                        }
+                        if (this.core.getPointerPath(child, 'src') === null) {
+                            violations.push({
+                                node: node,
+                                message: 'Src of connector [' + childPath + '] is null'
+                            });
+                        }
+
+                        if (this.core.getPointerPath(child, 'dst') != null) {
+                            var state = nodes[this.core.getPointerPath(child, 'dst')];
+                            var stateName = this.core.getAttribute(state, 'name');
+                            stateswithvalidTransitions[stateName]= this.core.getPath(state);
+                        }
+
+                        if (this.core.getPointerPath(child, 'src') != null) {
+                            var state = nodes[this.core.getPointerPath(child, 'dst')];
+                            var stateName = this.core.getAttribute(state, 'name');
+                            stateswithvalidTransitions[stateName]= this.core.getPath(state);
+                        }
+
+                        var expression = this.core.getAttribute(child, 'guardName');
+                        if(expression!=''){
+                            guardExpressions.push(expression);
+                        }
+
+                        var transitionMethod = this.core.getAttribute(child, 'transitionMethod');
+                        if (transitionMethod === '') {
+                            violations.push({
+                                node: node,
+                                message: 'transitionMethod empty for ' + childName + ' with path ' + childPath
+                            });
+                        }
+
+                        if (transitionNames.hasOwnProperty(childName)) {
+                            violations.push({
+                                node: node,
+                                message: 'Duplicated transition [' + childName + '] shared with ' + transitionNames[childName]
+                            });
+                        }
+                        transitionNames[childName] = this.core.getPath(child);
+                        //this.logger.info("*******",stateNames[childPath]);
+                    }
+
+                    if (this.isMetaTypeOf(child, this.META.InternalTransition)) {
+
+                        if (this.core.getPointerPath(child, 'dst') === null) {
```

Comment:
Venkat, just make sure that you do not check for name uniqueness in the internal transitions.